github_workflow.md

```diff
@@ -0,0 +1,153 @@
+label schema/taxonomy.
+
+First, every label should have one of four prefixes:
+
+1. `type:`
+2. `component:`
+3. `cat:`
+4. `status:`
+
+All issues of a given prefix should have the same color to make visual
+identification easy.
+
+### `type:`
+
+The `type:` prefix indicates the type of issue or pull request. There are four
+types:
+
+* `type:bug`
+* `type:enhancement`
+* `type:maintenance`
+* `type:question`
+
+Guidelines:
+
+* Each issue or pull request should have only one `type:` prefixed label.
+* Limiting ourselves to these four types allows us to query our issues in a 
+  structural and relational manner.
+
+### `component:`
+
+The `component:` prefix indicates the component or part of the software the
+issue is related to. Thus, on the JupyterLab repository, we have labels such as:
+
+* `component:About`
+* `component:Application`
+* `component:Build System`
+* `component:Command Palette`
+* `component:Completer`
+* `component:Console`
+* `component:Notebook`
+* `component:Terminal`
+* `component:File Browser`
+* `component:File Editor`
+* ...
+
+Guidelines:
+
+* A single issue or pull request can have multiple `component:` labels, but
+  usually it will be a small number.
+* Each subproject will have its own unique set of `component:` labels.
+
+### `cat:`
+
+The `cat:` prefix indicates the "category" of the issue. The category is sort of
+mixin to the `component:` label to indicate cross-cutting categories such as:
+
+* `cat:Accesibility`
+* `cat:Design and UX`
+* `cat:Documentation`
+* `cat:Examples`
+* `cat:New Contributor`
+* `cat:Settings`
+
+Guidelines:
+
+* A single issue or pull request can have multiple `cat:` labels.
+* Each subproject will have its own unique set of `cat:` labels, but effort
+  should be taken to use common names.
+
+### `status:`
+
+The `status:` prefix indicates the status of the issue or pull request, such as:
+
+* `status:Blocked`
+* `status:Duplicate`
+* `status:Help Wanted`
+* `status:Inactive`
+* `status:Needs Discussion`
+* `status:Needs Info`
+* `status:Needs Review`
+* `status:Needs User Testing`
+* `status:Wont Fix`
+* `status:Work in Progress`
+
+Guidelines:
+
+* A single issue or pull request can have multiple `status:` labels.
+* Subprojects should use ths standard set of `status:` labels, but are free to
+  add additional ones.
+
+## Structured queries 
+
+This taxonomy has been designed to allow structured queries to be performed on
+the issues of one or more repositories. An example of this type of query is
+provided below for the JuptyerLab repository. This visualization shows the
+number of issues grouped by type (color), component (x) and milestone (row):
+
+![JupyterLab Issues:](jupyterlab_issues.png)
+
+## Checklist
+
+Here is a brief checklist developers can use to help them follow this workflow:
+
+* Each issue/PR has one milestone or the `status:Needs Info` label.
+* Each issue/PR has exactly one `type:` label.
+* Each issue/PR has a few `component:` labels.
```

Comment:
> I am guessing that is how, in practice, "bug" labeled issues end up representing smaller amounts of work.

I tend to disagree as there are many more "bugs", with lot of interaction that you have to take into account than "enhancement".

The fuzziness bug/enhancement is due to the fact that during development time you can have "bugs" to not-yet-release feature. Which are the kind of "bugs" that are easy to work on as you are in the process of writing this feature and testing it. So changing things is 'easy'.

So I tend to be on Min's side.

Also "bug" convey to me: "It has to be fixed". Enhancement: it can slips through.
Speaking of "small amount of work" is immediately a red flag. it's the same red flag as conflating "simple" and "easy". It's not because it can take you or me 5 minutes to do something that it is true for everyone. 

Go sit with a new contributor, they will likely spend 40 minutes just to _find_ where to fix the bug.

Your example with webpack is an exception as well. The opened issue happen to need an architectural change. If you think all the developer of Jupyter and IPython spend the majority of their time doing architectural design, then you have lost touch with a couple us us who wouldlike to have the chance to think architecture more.